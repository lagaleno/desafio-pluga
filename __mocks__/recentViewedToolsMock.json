[
    {
      "app_id": "omie",
      "name": "Omie",
      "color": "#001E27",
      "icon": "https://assets.pluga.co/apps/icons/omie/omie-icon.svg",
      "link": "https://pluga.co/ferramentas/omie/"
    },
    {
      "app_id": "hotmart",
      "name": "Hotmart",
      "color": "#F04E23",
      "icon": "https://assets.pluga.co/apps/icons/hotmart/hotmart-icon.svg",
      "link": "https://pluga.co/ferramentas/hotmart/"
    },
    {
      "app_id": "eduzz",
      "name": "Eduzz",
      "color": "#FFCD33",
      "icon": "https://assets.pluga.co/apps/icons/eduzz/eduzz-icon.svg",
      "link": "https://pluga.co/ferramentas/eduzz/"
    }
]