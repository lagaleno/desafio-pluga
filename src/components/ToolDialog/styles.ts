import styled from "styled-components";

export const ToolDetailsStyles = styled.div`
    .iconStyle {
        background-color: ${props => props.color};
        border-radius: 50%;
    }

`;

export const RecentlyViewedToolsStyles = styled.div`
    margin-top: 1em;
`;