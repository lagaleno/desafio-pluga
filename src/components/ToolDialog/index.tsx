import React from 'react'

// Interface Import
import { ITool } from '@/interfaces/ITool';

// Material UI Components import
import { Box, Button, Dialog, DialogContent, DialogContentText, DialogTitle, Grid, IconButton, Modal, Typography, useMediaQuery, useTheme } from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';

// Components import
import ToolsCard from '../Card';

// Styles import
import Image from 'next/image';
import { RecentlyViewedToolsStyles, ToolDetailsStyles } from './styles';


interface IProp {
    open: boolean;
    handleCloseModal: any;
    tool: ITool;
    recentlyViewedTools: ITool[];
}


const mountRecentViewedToolsList = (list: ITool[]) => {
    if (list.length >= 1) {
        return (
            list.map((tool) => (
                <Grid key={tool.app_id} item xs={12} sm={4} md={4}>
                    {/* Se cliclou chama handleCloseModal*/}
                    <ToolsCard tool={tool} />
                </Grid>
            ))
        )
    }
    return (
        <>
            <Typography variant="h6"> Descubra mais sobre outras ferramentas clicando nos cartões  </Typography>
        </>
    )
}

const mountToolDetails = (tool: ITool, fullScreen: boolean) => {
    return (
        <Grid container rowSpacing={1} columnSpacing={5} justifyContent="center">
            <Grid item xs={12} sm={6} md={6} style={{ textAlign: `${fullScreen ? "center" : "right"}` }}>
                <Image width={150} height={150} src={tool.icon} alt={tool.name} className="iconStyle" />
            </Grid>
            <Grid item xs={12} sm={6} md={6} style={{ textAlign: `${fullScreen ? "center" : "left"}` }}>
                <Typography variant="h6"> {tool.name} </Typography>
                <Button target="_blank" variant="contained" href={tool.link}>Acessar</Button>
            </Grid>
        </Grid>
    )
}

const ToolDialog = ({ open, handleCloseModal, tool, recentlyViewedTools }: IProp): JSX.Element => {
    const theme = useTheme();
    const fullScreen = useMediaQuery(theme.breakpoints.down('md'));


    return (
        <Dialog
        fullScreen={fullScreen}
        maxWidth={"md"}
        fullWidth={true}
        open={open}
        onClose={handleCloseModal}
        aria-labelledby="responsive-dialog-title"
      >
        <DialogTitle id="responsive-dialog-title" style={{ textAlign: "right" }}>
            <IconButton onClick={handleCloseModal}>
                <CloseIcon />
            </IconButton>
        </DialogTitle>
        <DialogContent>
            <ToolDetailsStyles color={tool.color}>
                {
                    mountToolDetails(tool, fullScreen)
                }
            </ToolDetailsStyles>
            <RecentlyViewedToolsStyles>
                <Grid container rowSpacing={2} columnSpacing={5} justifyContent="center">
                    <Grid item xs={12} style={{ textAlign: `${fullScreen ? "center" : "left"}` }}>
                        <Typography variant="h5"> Últimas Ferramentas Visualizadas </Typography>
                    </Grid>
                    {
                        mountRecentViewedToolsList(recentlyViewedTools)
                    }
                </Grid>
            </RecentlyViewedToolsStyles>
        </DialogContent>
      </Dialog>
    )

}

export default ToolDialog;