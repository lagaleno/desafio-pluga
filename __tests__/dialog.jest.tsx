// Jest imports
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'intersection-observer';

// Component import
import ToolDialog from '@/components/ToolDialog';

// Mock import
import toolsMock from "../__mocks__/toolsMock.json";
import recentViewedToolsMock from "../__mocks__/recentViewedToolsMock.json"

// { open, handleCloseModal, tool, recentlyViewedTools }
describe ('Tool Dialog', () => {
    it ('Renders component correctly', () => {
        toolsMock.forEach((tool) => {
            const { container } = render (<ToolDialog open handleCloseModal={jest.fn()} tool={tool}  recentlyViewedTools={recentViewedToolsMock} />)
            expect(container).toMatchSnapshot()
        })           
    })
})