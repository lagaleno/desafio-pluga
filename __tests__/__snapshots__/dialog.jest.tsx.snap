// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool Dialog Renders component correctly 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 2`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 3`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 4`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 5`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 6`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 7`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 8`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 9`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 10`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 11`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 12`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 13`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 14`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 15`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 16`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 17`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 18`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 19`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 20`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 21`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 22`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 23`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 24`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 25`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 26`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 27`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 28`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 29`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 30`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 31`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 32`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 33`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 34`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 35`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 36`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 37`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 38`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 39`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 40`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 41`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 42`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 43`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 44`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 45`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 46`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 47`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 48`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 49`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 50`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 51`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 52`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 53`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 54`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 55`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 56`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 57`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 58`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 59`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 60`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 61`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 62`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 63`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 64`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 65`] = `
<div
  aria-hidden="true"
/>
`;

exports[`Tool Dialog Renders component correctly 66`] = `
<div
  aria-hidden="true"
/>
`;
